webapp/channels/src/components/announcement_bar/notification_permission_bar/index.test.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {screen, fireEvent, act} from '@testing-library/react';
+import React from 'react';
+
+import {renderWithContext} from 'tests/react_testing_utils';
+import {requestNotificationPermission, isNotificationAPISupported} from 'utils/notifications';
+
+import NotificationPermissionBar from './index';
+
+jest.mock('utils/notifications', () => ({
+    requestNotificationPermission: jest.fn(),
+    isNotificationAPISupported: jest.fn(),
+}));
+
+describe('NotificationPermissionBar', () => {
+    const initialState = {
+        entities: {
+            users: {
+                currentUserId: 'user-id',
+            },
+            general: {
+                config: {},
+                license: {},
+            },
+        },
+    };
+
+    beforeEach(() => {
+        (isNotificationAPISupported as jest.Mock).mockReturnValue(true);
+        (window as any).Notification = {permission: 'default'};
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        delete (window as any).Notification;
+    });
+
+    test('should render the notification bar when conditions are met', () => {
+        const {container} = renderWithContext(<NotificationPermissionBar/>, initialState);
+
+        expect(container).toMatchSnapshot();
```

Comment:
Do you think we need these snapshots? If not, I'd recommend removing them since we're testing the functionality of the component already, and it helps signal to people that we have better ways of testing with Testing Library

Reply:
removed the snapshots